src/NServiceBus.Testing.Tests/Saga/ExpectReplyToOriginatorTests.cs

```diff
@@ -0,0 +1,93 @@
+﻿namespace NServiceBus.Testing.Tests.Saga
```

Comment:
I added these (partially failing) tests as some of the logic was using GetCorrelationId to differentiate between `context.Reply` and `ReplyToOriginator` which internally both use `context.Reply` but ReplyToOriginator also sets a specific CorrelationId which we now no longer can access to differentiate between the scenarios. The new behavior is essentially to verify the destination of the reply:
* if the destination is set to the "default" originator, it both matches the `ExpectReplyToOriginator` as well as the `ExpectReply` expectations
* if the reply destination is set explicitly to a value using the `replyOptions.SetDestination` API, `ExpectReplyToOriginator` will fail